cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/java/co/cask/cdap/report/main/MessageUtil.java

```diff
@@ -185,7 +186,7 @@ private static String getLatestMessageId(List<Location> locationsTimeSorted) thr
       Location latestLocation = locationsTimeSorted.get(index);
       try {
         DataFileReader<GenericRecord> dataFileReader =
-          new DataFileReader<>(new File(latestLocation.toURI()),
+          new DataFileReader<>(new LocationSeekableInput(latestLocation),
```

Comment:
I don't understand why we need a SeekableInput. Is it because we can't use a File or is there a real reason we need SeekableInput?

If we're just trying to avoid using File, can you use DataFileStream using the location's input stream?

Reply:
yes, we can't use file, would not work in cluster. However we can use `DataFileStream`

Having seekable input will help to skip to end of the file, as we are only interested in the last record. however that perf improvement is not implemented yet. 

so its better to use `DataFileStream` now, ill undo other changes.